server/app/controllers/ti/TrustedIntermediaryController.java

```diff
@@ -195,26 +250,31 @@ public Result updateClientInfo(Long id, Http.Request request) throws ApplicantNo
     if (trustedIntermediaryGroup.isEmpty()) {
       return unauthorized();
     }
-    Form<EditTiClientInfoForm> form =
-        formFactory.form(EditTiClientInfoForm.class).bindFromRequest(request);
+    Form<TiClientInfoForm> form = formFactory.form(TiClientInfoForm.class).bindFromRequest(request);
     form =
         tiService.updateClientInfo(
             form, trustedIntermediaryGroup.get(), id, messagesApi.preferred(request));
     if (form.hasErrors()) {
       return ok(
           editTiClientView.render(
-              trustedIntermediaryGroup.get(),
-              ApplicantPersonalInfo.ofLoggedInUser(
+              /* tiGroup= */ trustedIntermediaryGroup.get(),
+              /* personalInfo= */ ApplicantPersonalInfo.ofLoggedInUser(
                   Representation.builder()
                       .setName(
                           form.value().get().getFirstName()
                               + " "
                               + form.value().get().getLastName())
                       .build()),
-              request,
-              messagesApi.preferred(request),
-              id,
-              Optional.of(form)));
+              /* request= */ request,
+              /* messages= */ messagesApi.preferred(request),
+              /* accountIdToEdit= */ Optional.of(id),
+              /* applicantIdOfTi= */ civiformProfile
+                  .get()
+                  .getApplicant()
+                  .toCompletableFuture()
+                  .join()
+                  .id,
```

Comment:
suggestion (non-blocking): It looks like you fetch the `applicantIdOfTi` the same way in a lot of these methods. Is it worthwhile to have a helper method that does it that can be reused?

Reply:
Moved to a helper. :)